awx/main/models/inventory.py

```diff
@@ -1404,5 +1405,10 @@ def get_absolute_url(self, request=None):
         return reverse('api:inventory_script_detail', kwargs={'pk': self.pk}, request=request)
 
 
-for cls in PluginFileInjector.__subclasses__():
-    InventorySourceOptions.injectors[cls.__name__] = cls
+awx_entry_points = {ep.name: ep for ep in entry_points(group='awx_plugins.inventory')}
+supported_entry_points = {ep.name: ep for ep in entry_points(group='awx_plugins.inventory.supported')}
+entry_points = awx_entry_points if detect_server_product_name() == 'AWX' else {**awx_entry_points, **supported_entry_points}
+
+for entry_point_name, entry_point in entry_points.items():
+    cls = entry_point.load()
+    InventorySourceOptions.injectors[entry_point_name] = cls
```

Comment:
It'd be best if this wasn't happening during the import time but called in some predictable hook point, early in django's lifecycle.

Reply:
Yep, [here](https://github.com/ansible/awx/blob/cb04ad8ef5b5423ecf376925e77f4d3247af1cd9/awx/main/apps.py#L73) is a better place